ark/jsonschema/json.ts

```diff
@@ -0,0 +1,80 @@
+import type { JsonSchemaOrBoolean } from "@ark/schema"
+import { printable, throwParseError } from "@ark/util"
+import { type, type JsonSchema } from "arktype"
+import { parseArrayJsonSchema } from "./array.ts"
+import { parseCommonJsonSchema } from "./common.ts"
+import { parseCompositionJsonSchema } from "./composition.ts"
+import { parseNumberJsonSchema } from "./number.ts"
+import { parseObjectJsonSchema } from "./object.ts"
+import { JsonSchemaScope } from "./scope.ts"
+import { parseStringJsonSchema } from "./string.ts"
+
+const jsonSchemaTypeMatcher = type.match
+	.in<JsonSchema>()
+	.at("type")
+	.match({
+		"unknown[]": jsonSchema =>
+			parseCompositionJsonSchema({
+				anyOf: jsonSchema.type.map(t => ({ type: t as never }))
+			}),
+		"'array'": jsonSchema => parseArrayJsonSchema.assert(jsonSchema),
+		"'boolean'|'null'": jsonSchema => type(jsonSchema.type),
+		"'integer'|'number'": jsonSchema =>
+			parseNumberJsonSchema.assert(jsonSchema),
+		"'object'": jsonSchema => parseObjectJsonSchema.assert(jsonSchema),
+		"'string'": jsonSchema => parseStringJsonSchema.assert(jsonSchema),
+		default: () => undefined
+	})
+
+export const innerParseJsonSchema = JsonSchemaScope.Schema.pipe(
+	(jsonSchema: JsonSchemaOrBoolean): type.Any => {
+		if (typeof jsonSchema === "boolean") {
+			if (jsonSchema) return JsonSchemaScope.Json
+			else return type.never // No runtime value ever passes validation for JSON schema of 'false'
+		}
+
+		if (Array.isArray(jsonSchema)) {
+			return (
+				parseCompositionJsonSchema({ anyOf: jsonSchema }) ??
+				throwParseError(
```

Comment:
you can avoid handling this case by exporting `parseAnyOfSchema` and just passing it directly there

Reply:
Good catch; done in 886d2db9